added support for gfx1151 in -a default parameter in rmake.py

NguyenNhuDi opened the pull request.

NguyenNhuDi committed:
added support for gfx1151 in -a default parameter in rmake.py

TorreZuk reviewed (commented):
The original rmake for windows building did not include the notion of default gfx lists so I am not clear if we are adding some potential for mis-matches across releases with this explicit list.  I see one pre-existing list was already present so this is just a comment for consideration.

spolifroni-amd committed:
added link to documentation (#623)

NguyenNhuDi committed:
merge back 6.3 hotfixes to develop (#634)

* Remove website URL from comments (#600)

Referencing or using code from some websites is prohibited in this repository.
This change removes an informational reference in the comments.

* Fix rare memory access faults when using internal serial merge (#597)

* test: add tests for internal serial merge function

* refactor(detail/merge_path.hpp): removed code duplication

* fix(detail/merge_path.hpp): stricter boundary checking in serial merge

* fix(detail/block_sort_merge.hpp): fix missing block-wide sync

During a previous refactor, serial_merge does no longer do a block sync. This has now been re-added.

* feat: add unsafe variant of serial merge

* fix: use bounded version for serial merge to fix rare page faults

* test(test_internal_merge_path): clean up internal merge path tests

* style: standardize range_t<> construction

* fix(detail/merge_path.hpp): fix 'range_t<>::count1()' and 'range_t<>::count2()' return types to be same as encapsulated type

* perf(detail/merge_path.hpp): use const ref in function parameters

* refactor(detail/merge_path.hpp): replace redundant use of 'OffsetT' with 'unsigned int'

* chore: update changelog

* fix: restore missing thread sync

This got removed during a rebase.

* Add gfx1151 target (#601) (#603)

Co-authored-by: Stanley Tsang <stanley.tsang@amd.com>

* Merge back 6.2 hotfixes (#607) (#620)

* Update dependency names for static builds (#557)

This also removes the line setting `BUILD_SHARED_LIBS` to `ON`, which was previously required to get the correctly named packages when not specifically compiling for a static build. Updates to the ROCmCMakeBuildTools (rocm-cmake) should mean this is no longer necessary.

* Fix BUILD_SHARED_LIBS for packaging (#558)

* Fix the dependencies of the static packages (#563)

* cmake: don't set CMAKE_C_COMPILER, as rocPRIM is a CXX project (#567)

* add developer guidelines (#555) (#574)



* Update Read the Docs config to Python 3.10 and latest rocm-docs-core (#564) (#579)

* Cherry-pick: Optimize block_reduce_warp_reduce when block size is the same as warp size (#599)

* Optimize block_reduce_warp_reduce when block size == warp size

* Make conditional constexpr

* Fix conflict in concepts.rst

---------

Co-authored-by: Lauren Wrubleski <Lauren.Wrubleski@amd.com>
Co-authored-by: Steve Leung <Steve.Leung@amd.com>
Co-authored-by: randyh62 <42045079+randyh62@users.noreply.github.com>
Co-authored-by: Nol Moonen <nol@streamhpc.com>
Co-authored-by: Sam Wu <22262939+samjwu@users.noreply.github.com>

* Changed precondition for edge case in serial_merge to prevent assertion error (#622)

* added std::min to ensure no out of bound acess

* fixed typo keys->keys1

* updated changelog

* reverted std::min

* implemented suggested logic

* edited to conform to standards (#618)

* Memory leak fix for multiple rocPRIM unit  tests (#614)

* fixed mem leak in test_config_dispatch.cpp

* added missing hip free for method==4 in test_block_scan.kernels

* added graphHelpeer class that does not cause memory leak due to using hipGraphCreate

* replaced old hipGraph helpers with new class in device_bin_search

* changed HIP_CHECK_NON_VOID to HIP_CHECK

* fixed mem leak in device_bin_search

* added additional functions

* changed out old calls to hipGraphCrete to new GraphHelper class

* added missing stream sync for hipgrag_algs

* n

* added missing hipFree and HIP_CHECK for lookback_reproducibility

* added missing hipFree in test_discard_iterator

* fixed test failures

* removed extra hipFree

* removed unused variables

* updated change log

* removed redundant function

---------

Co-authored-by: Your Name <you@example.com>
Co-authored-by: root <root@x1001c6s0b1n0.hostmgmt2001.cm.lockhart.amd.com>

* changed to using gHelper

* merged 6.3 version of merge_sort_merge_path (failing large indices test)

* updated mergepath to have fix for large indices

* updated the changelog for 6.3 (#632)

* updated changelog for test_utils_hipgraph change

* updated changelog

---------

Co-authored-by: Wayne Franz <wayfranz@amd.com>
Co-authored-by: Nara <nara@streamhpc.com>
Co-authored-by: amd-garydeng <garydeng@amd.com>
Co-authored-by: Stanley Tsang <stanley.tsang@amd.com>
Co-authored-by: Lauren Wrubleski <Lauren.Wrubleski@amd.com>
Co-authored-by: Steve Leung <Steve.Leung@amd.com>
Co-authored-by: randyh62 <42045079+randyh62@users.noreply.github.com>
Co-authored-by: Nol Moonen <nol@streamhpc.com>
Co-authored-by: Sam Wu <22262939+samjwu@users.noreply.github.com>
Co-authored-by: spolifroni-amd <Sandra.Polifroni@amd.com>
Co-authored-by: Your Name <you@example.com>
Co-authored-by: root <root@x1001c6s0b1n0.hostmgmt2001.cm.lockhart.amd.com>

umfranzw commented on rmake.py:
```diff
@@ -124,7 +127,10 @@ def config_cmd():
     cmake_options.append( tools )
 
-    cmake_options.extend( cmake_platform_opts)
+    if os.name == 'nt':
```
It looks like the reason that cmake_platform_opts is a sometimes a string instead of a list is because of line 125:
`cmake_platform_opts = f"-DROCM_DIR:PATH={rocm_path} -DCPACK_PACKAGING_INSTALL_PREFIX={rocm_path}"`

Could we change that line to:
`cmake_platform_opts.append(f"-DROCM_DIR:PATH={rocm_path} -DCPACK_PACKAGING_INSTALL_PREFIX={rocm_path}")`

I think that would make sure that it's always a list, regardless of what platform we're on. Then we wouldn't need the new if/else block that was added here (we could just go back to the original code):
`cmake_options.extend(cmake_platform_opts)`
Feel free to let me know what you think.

umfranzw replied:
As you've mentioned in the other similar changes, keeping the options separate in the list is probably a good idea, eg:
cmake_platform_opts = [f"-DROCM_DIR:PATH={rocm_path}", f"-DCPACK_PACKAGING_INSTALL_PREFIX={rocm_path}"]

NguyenNhuDi replied:
Updated cmake_platform_opts!

NguyenNhuDi committed:
added missing gfx12 in config (#633)

NguyenNhuDi committed:
changed cmake_platform_opts to always be a list

macurtis-amd committed:
Fix clang compilation error: missing-template-arg-list-after-template-kw (#627)

Full text of error:

  a template argument list is expected after a name prefixed by the template
  keyword [-Wmissing-template-arg-list-after-template-kw]

Found while testing a recent build of clang 20, though I think clang 19 will
also complain.

NguyenNhuDi committed:
Merge branch 'develop' into zenguyen/rmake-support
